### PR TITLE
Update Vert.x quickstart code to match the new guide.

### PR DESCRIPTION
**Pull request by cescoffier**
See https://github.com/quarkusio/quarkus/pull/18453.

**Checklist**:

Your pull request:

- [X] targets the `development` branch
- [X] uses the `999-SNAPSHOT` version of Quarkus
- [X] has tests (`mvn clean test`)
- [X] works in native (`mvn clean package -Pnative`)
- [X] has native tests (`mvn clean verify -Pnative`)
- [X] makes sure the associated guide must not be updated
- [X] links the guide update pull request (if needed)
- [X] updates or creates the `README.md` file (with build and run instructions)

### CODE REVIEW

File: vertx-quickstart/pom.xml [hunk 1]
```diff
@@ -113,14 +115,9 @@
...
                 </property>
             </activation>
-            <properties>
-                <quarkus.package.type>native</quarkus.package.type>
-                <!-- <quarkus.native.enable-https-url-handler>true</quarkus.native.enable-https-url-handler> -->
-            </properties>
             <build>
                 <plugins>
                     <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
```
Comment (geoand) [inline vertx-quickstart/pom.xml:123]: I would likely leave these as IntellJ shows false errors when they are missing

File: vertx-quickstart/src/main/java/org/acme/VertxResource.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+@Path("/vertx")                        // <1>
+public class VertxResource {
+
+    private final Vertx vertx;
+    private final WebClient client;
+
+    @Inject
```
Comment (geoand) [inline vertx-quickstart/src/main/java/org/acme/VertxResource.java:23]: Can be ommited

File: vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java [hunk 3]
```diff
@@ -32,40 +17,39 @@
...
+    private void initdb(@Observes StartupEvent ev) {
         if (schemaCreate) {
-            initdb();
+            client.query("DROP TABLE IF EXISTS fruits").execute()
+                    .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)")
+                            .execute())
+                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')").execute())
+                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')").execute())
+                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')").execute())
+                    .await().indefinitely();
```
Comment (geoand) [inline vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java:49]: @edeandrea this is similar to what you were asking me the other day

File: vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java [hunk 4]
```diff
@@ -75,7 +59,7 @@ private void initdb() {
 
     @GET
     @Path("{id}")
-    public Uni<Response> getSingle(@PathParam Long id) {
+    public Uni<Response> getSingle(@RestPath Long id) {
```
Comment (geoand) [inline vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java:62]: The `@RestPath` annotation can be ommited entirely

File: vertx-quickstart/src/main/java/org/acme/extra/ResourceUsingWebClient.java [hunk 5]
```diff
@@ -29,7 +26,7 @@ public ResourceUsingWebClient(Vertx vertx) {
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/{name}")
-    public Uni<JsonObject> getFruitData(@PathParam("name") String name) {
+    public Uni<JsonObject> getFruitData(@RestPath("name") String name) {
```
Comment (geoand) [inline vertx-quickstart/src/main/java/org/acme/extra/ResourceUsingWebClient.java:29]: Same here

**Review by geoand — COMMENTED** [review]
Thanks for this!

I added a few comments

File: vertx-quickstart/pom.xml (hunk 1 above)
Comment (cescoffier) [reply vertx-quickstart/pom.xml:123]: Hum... @ia3andy I believe code.quarkus.io does not contain the groupId. Is it something we should fix there too?

Comment (geoand) [reply vertx-quickstart/pom.xml:123]: I think there is an open issue about that somewhere...

File: vertx-quickstart/src/main/java/org/acme/extra/ResourceUsingWebClient.java (hunk 5 above)
Comment (cescoffier) [reply vertx-quickstart/src/main/java/org/acme/extra/ResourceUsingWebClient.java:29]: Didn't know that! Nice!

**Comment by cescoffier** [comment]
@geoand Removed all the RestPath

**Review by geoand — APPROVED** [review]

File: vertx-quickstart/pom.xml (hunk 1 above)
Comment (ia3andy) [reply vertx-quickstart/pom.xml:123]: Yeah it's been fixed, let me find the commit

Comment (ia3andy) [reply vertx-quickstart/pom.xml:123]: https://github.com/quarkusio/quarkus/commit/508639444b4f7da9711382136cc637b62037b4a0

File: vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java (hunk 3 above)
Comment (edeandrea) [reply vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java:45]: Yeah I was looking for a way to do initialization like this in a non-blocking way.

**Pull request merged**